src/electrum/server.rs

```diff
@@ -490,6 +493,27 @@ impl Connection {
         Ok(result)
     }
 
+    fn log_rpc_event(&self, entries: &Vec<(&str, Value)>) {
```

Comment:
This should ideally avoid preparing and allocating the log `entries` when logging is turned off to avoid the unnecessary performance hit.

I would consider using a macro to make `entries` evaluate only when logging is turned on. Alternatively this can be lazily evaluated by passing a thunk closure that computes `entries`, or simply by wrapping the outer calls in an `if`.

Reply:
I've used macro that creates log entries only if `--electrum-rpc-logging` flag is set (see latest [commit](bfa5b9f84e8fcbb77da9f7f6d8b3720bfb683250))